src/view/dataElements/identityMap.jsx

```diff
@@ -74,6 +74,30 @@ const validateDuplicateValue = (
   );
 };
 
+const hasPrimary = identifier => identifier.primary;
+
+const validateOnePrimary = (createError, identities, message) => {
+  let primaries = 0;
+
+  for (let i = 0; i < identities.length; i += 1) {
+    const { identifiers = [] } = identities[i];
+
+    for (let j = 0; j < identifiers.length; j += 1) {
+      if (hasPrimary(identifiers[j])) {
+        primaries += 1;
+      }
+
+      if (primaries > 1) {
+        return createError({
```

Comment:
Ideally, we would flag multiple primary checkboxes as invalid. For example, if three identifiers had primary checked, the 2nd and 3rd ones should be marked invalid. I think you can return an array of errors.

Reply:
I did it the way we check for duplicates, we show error for the first duplicate we found. If I change this one I think we should change the other validators as well. Either one works fine for me, I just want to be consistent.